module/observations/helper.js

```diff
@@ -2688,4 +2711,134 @@ module.exports = class ObservationsHelper {
+              success: true
+            })
+          }
+
+          let roles = bodyData.role.split(",");
+
+          let roleArray = [];
+          for(let i=0;i<roles.length;i++){
+
+          let rolesDocumentAPICall = await entityManagementService.userRoleExtension({
+            code: roles[i],
+          },
+          ["entityTypes.entityType"])
+
+          if(rolesDocumentAPICall.success){
+            roleArray.push(rolesDocumentAPICall.data[0].entityTypes[0].entityType)
+          }
+
+        }
+
+          const uniqueroleArray = _.uniq(roleArray);
+          let filterData = {
+            _id:topLevelEntityId,
+            entityType: topLevelEntityType,
+            deleted:false
+           };
+         
+          //Retrieving the entity from the Entity Management Service
+           let entitiesDocument = await entityManagementService.entityDocuments(
+             filterData
+           );
+
+           if (!entitiesDocument.success) {
+             throw {
+               status: httpStatusCode.bad_request.status,
+               message: messageConstants.apiResponses.ENTITIES_NOT_FOUND,
+             };
+           }
+
+          let childHierarchyPath = entitiesDocument.data[0].childHierarchyPath;
+          childHierarchyPath.unshift(topLevelEntityType)
+
+          const highestEntityType = this.findHighestHierarchy(uniqueroleArray,childHierarchyPath);
+          const highestIndex = childHierarchyPath.indexOf(highestEntityType);
+          const solutionIndex = childHierarchyPath.indexOf(solutionEntityType);
+        
+          if (highestIndex === -1 || solutionIndex === -1) {
+            throw {
+              status: httpStatusCode.bad_request.status,
+              message: messageConstants.apiResponses.INVALID_ENTITY_TYPE
+          };   
+          }
+        
+          if (highestIndex > solutionIndex) {
+            throw {
+              status: httpStatusCode.bad_request.status,
+              message: messageConstants.apiResponses.OBSERVATION_NOT_RELEVENT_FOR_USER
+          };
+        
+        }
+
+        resolve({
+          success: true
+        })
+
+        } catch (error) { 
+          return resolve({
+            status: error.status || httpStatusCode.internal_server_error.status,
+            message:
+              error.message || httpStatusCode.internal_server_error.message,
+            data: false,
+          });
+        }
+      });
+    }
+      /**
+   * Get the highest role in the heirarchy.
+   * @method
+   * @name findHighestHierarchy
+   * @param {String} roles - Roles of the user.
+   * @param {Array} rolesHierarchy - Roles hierarchy in descending order.
+   * @returns {String} returns highest role in the heirarchy.
+   */
+    static findHighestHierarchy(roles,rolesHierarchy) {
+      
+        let highestHierarchyValue = null;
+        let highestIndex = Infinity;
+      
+        // Loop through each role in the given array
+        roles.forEach(role => {
```

Comment:
@borkarsaish65 this is  not  role, it is entityTypes associated with the role - change variable name